src/webots/gui/WbDragTransformEvent.cpp

```diff
@@ -80,16 +83,41 @@ WbDragHorizontalEvent::~WbDragHorizontalEvent() {
 }
 
 void WbDragHorizontalEvent::apply(const QPoint &currentMousePosition) {
-  mViewpoint->viewpointRay(currentMousePosition.x(), currentMousePosition.y(), mMouseRay);
-  mDragPlane.redefine(mUpWorldVector, mSelectedTransform->position());
-  mIntersectionOutput = mMouseRay.intersects(mDragPlane);
-  WbVector3 displacementFromInitialPosition =
-    mMouseRay.point(mIntersectionOutput.second) + mTranslationOffset - mInitialPosition;
-  // remove any x or z scaling from parents (we shouldn't touch y as we're moving on the world horizontal plane)
-  displacementFromInitialPosition.setX(displacementFromInitialPosition.x() / mScaleFromParents.x());
-  displacementFromInitialPosition.setZ(displacementFromInitialPosition.z() / mScaleFromParents.z());
-  mSelectedTransform->setTranslation((mInitialPosition + displacementFromInitialPosition).rounded(WbPrecision::GUI_MEDIUM));
-  mSelectedTransform->emitTranslationOrRotationChangedByUser();
+  if (!mIsInitializationDone) {
+    WbRay normalizedMouseRay = mMouseRay;
+    normalizedMouseRay.normalize();
+    // event occurs only if the mouse ray is not parallel to the horizontal drag plane
+    if (abs(normalizedMouseRay.direction().dot(mUpWorldVector)) > DRAG_HORIZONTAL_MIN_COS)
+      mIsMouseRayValid = true;
+    else {
+      mIsMouseRayValid = false;
+      WbLog::warning(tr("To drag this element, rotate first the view so that the horizontal plane is clearly visible."));
```

Comment:
Maybe this is better
```suggestion
      WbLog::warning(tr("To drag this element, first rotate the view so that the horizontal plane is clearly visible."));
```